infrastructure/zk/src/init.ts

```diff
@@ -138,6 +140,92 @@ export async function submoduleUpdate() {
     await utils.exec('git submodule update');
 }
 
+function updateConfigFile(path: string, modeConstantValues: Record<string, number | string | null>) {
```

Comment:
This approach looks a little suspicious, like we're cleaning something up in a way that feels fragile and/or not maintainable (will we remember why we're doing all of this in a few months?). 

I haven't looked super carefully here but would like to understand what we're doing here better.